### ISSUE TITLE
lamppost model

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#181]
For reflection models we need to have a lamp post source model.  Presumably we want this to be described by a position above and below the disk, as is the case for the xion model

https://heasarc.gsfc.nasa.gov/xanadu/xspec/manual/XSmodelXion.html

in xspec

Presumably we want the lamp-post fo be symmetric above and below the accretion disk and to replace (as an option) our currently model which has the boundary layer radiation arising from the surface of the "star".

**Comment by jhmatthews** [comment]
So I guess we'd require an additional option in the file- if power law radiation is specified then we'd add something like

```
geometry_for_pl_source(0=sphere,1=lamp_post)  
```

and if that equals one then add somewhere the parameter

```
lamp_post.height(r_g)
```

And then simply put isotropic X-ray photons as a point source at that height?

**Comment by jhmatthews** [comment]
Had a quick look at the code and this is dead easy so I'll quickly code it. Is it something that should just go in domain2? It occurs to me it might be sensible to run a current AGN model with it and so put it in dev then merge into domain2? Out current X-ray source has an interesting "feature" whereby the X-ray source is weaker at low inclinations because photons are intercepted by the disk....

**Comment by kslong** [comment]
Yes, this should go in domain2

**Comment by jhmatthews** [comment]
Ok will do. I'm going to run a QSO model with it just to check if it affects the ionization state, say.

**Comment by jhmatthews** [comment]
This could probably do with testing, but is now merged into domain2 via pull request #184
